### ISSUE TITLE
gatsby v2.13.65 -> v2.13.66 introduces a lot of warnings in Windows due to path changes

### ISSUE THREAD

**Issue by karland** [gatsbyjs/gatsby#16721]
## Description

I am on Windows 10. After updating from Gatsby v2.13.65 -> 2.13.66 I get a lot of warnings resulting from the path letter being different, such as this

```
 warning  in D:/my/Websites/node_modules/object-assign/index.js
⠀
There are multiple modules with names that only differ in casing.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Use equal casing. Compare these module identifiers:
* D:\my\Websites\node_modules\object-assign\index.js
    Used by 160 module(s), i. e.
    D:\my\Websites\node_modules\react\cjs\react.development.js
* d:\my\Websites\node_modules\object-assign\index.js
    Used by 4 module(s), i. e.
    d:\my\Websites\node_modules\react\cjs\react.development.js
```

I believe the error results from https://github.com/gatsbyjs/gatsby/commit/cac4246

Up until Gatsby v2.13.65 all path letters were in lower cases, which was good.

I also believe the Windows path letters in packages/gatsby/src/utils/__tests__/path.js should be lower case.

**Comment by karland** [comment]
@waltercruz Did you see this issue? Thanks.

**Comment by waltercruz** [comment]
I will check it ou!

**Comment by caseyWebb** [comment]
See [this comment](https://github.com/Profiscience/true-case-path/issues/3#issuecomment-528832215).

tl;dr: use powershell

**Comment by waltercruz** [comment]
Hi @karland , have you seen @caseyWebb comment?

I still can implement his last suggestion, "_Gatsby could always enforce/assume uppercase drive letters when checking paths_", but I will need some more days (don't have access any windows machine to test).

**Comment by karland** [comment]
Hi @waltercruz. Yes, I have seen the excellent and very nice explanation of @caseyWebb. Based on this explanation, I now understand the source of the problem and can now work around it. I still think, however, it would be good in Gatsby internally to always convert Windows drive letters to uppercase to solve this problem for good, but I have no clue how difficult this is. Thanks a lot for your kind offer/help.

**Comment by jannikbuschke** [comment]
I'm seeing this issue in Azure Devops:
![image](https://user-images.githubusercontent.com/5894881/64474689-863df980-d178-11e9-9f08-dbb661b4b39f.png)
with NPM tasks

**Comment by waltercruz** [comment]
Hey @jannikbuschke , please see [this comment](https://github.com/Profiscience/true-case-path/issues/3). BTW, glad that you remembered of Azure, I will try to use a windows machine there to test. I will try to fix it in the next days.

**Comment by sidharthachatterjee** [comment]
Fixed and published in `gatsby@2.15.10` thanks to @waltercruz 🥇

**Comment by stephen-zhao** [comment]
Hello.. so it seems I am still getting a very similar issue. I am on Gatsby version 2.15.29 and Gatsby-CLI version 2.7.54, running on WSL Ubuntu.

```
warn The plugin "dev-404-page" created a page with a component path that doesn't match the casing of the actual file.
This may work locally, but will break on systems which are case-sensitive, e.g. most CI/CD pipelines.
page.component:     "/mnt/d/projects/Dev_Projects/stephen-zhao.github.io/.cache/dev-404-page.js"
path in filesystem: "/mnt/d/Projects/Dev_Projects/stephen-zhao.github.io/.cache/dev-404-page.js"
                            ^
```

The same applies to plugin `gatsby-plugin-page-creator` as well as several components inside `node_modules`. Not sure what is going on... Why would gatsby want to change the first uppercase character it encounters to a lowercase (the subsequent uppercase characters are unaffected)?

**Comment by waltercruz** [comment]
> Why would gatsby want to change the first uppercase character it encounters to a lowercase (the subsequent uppercase characters are unaffected)?

Long history, but it's about windows filesystem and the weird way that windows handles it. See [this comment](https://github.com/Profiscience/true-case-path/issues/3#issuecomment-528832215).

But long story short, it should not try to uppercase the first letter if it's running on WSL. My code by now just checks if is running on windows, I should probably check if running on WSL and don't fo the uppercase of the first letter.

I will create a ticket and assign it to myself, but I will only have time to do it in some days.
